syft/serde/msgpack/native_serde.py

```diff
@@ -364,6 +363,20 @@ def _detail_ndarray(
     return res
 
 
+def _simplify_type(worker: AbstractWorker, my_type: type) -> bytes:
+    module_path = my_type.__module__
+    full_path_type = module_path + "." + my_type.__name__
+    return full_path_type.encode("utf-8")
+
+
+def _detail_type(worker: AbstractWorker, type_represantation: bytes) -> type:
+    string_representation = type_represantation.decode("utf-8")
+    result = pydoc.locate(string_representation)
+    if result is None:
+        return object
```

Comment:
A generic object for the isinstance operation, currently I am doing this because I am using pydoc to recreate the original type (this assumes that both ends use the same file layout). If pydoc failes, this means that the specific type was not found and I want to drop typechecking on that element. This will go away in time, after the protobuf support becomes big enough and we can rely on the protobuf type (this is actually your idea :+1: ). Might add a warning and maybe a future TODO mark.